7 фз

Maks1-15 opened the pull request:
Добавлена функциональность связанная с файлами, сделаны методы для работы с файлом добавлено собственное исключение
в main написан функционал работы с файлом

Maks1-15 committed:
7 фз
Добавлена функциональность связанная с файлами, сделаны методы для работы с файлом
добавлено собственное исключение
в main написан функционал работы с файлом

Umka7389 commented on src/Main.java:
```diff
@@ -1,46 +1,49 @@
-        tm.createTask(task2);
-        tm.createEpic(epic1);
-        tm.createEpic(epic2);
+        fbtm.createTask(task1);
```
Стоит давать осмысленные названия переменным, чтобы сразу было понятно что это

Umka7389 commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,168 @@
+        Status status = Status.valueOf(arrayString[4]);
+
+        switch (type) {
+            case "TASK":
```
Напрашивается enum

Umka7389 commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -69,7 +69,7 @@ public Task getIdTask(int id) {
             return tasks.get(id);
```
Сейчас из названия метода ожидается, что он вернет id. Более подходящее getTaskById

Umka7389 commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -69,7 +69,7 @@ public Task getIdTask(int id) {
         }
         // Видел, что null не стоит возвращать, поэтому возвращаю объект
-        return new Task(null, null, Status.NULL);
+        return null;
```
Ну возвращать новый объект если вы не нашли по ид явно хуже) 
В данном случае map сама возвращает null, поэтому можно вынести tasks.get(id) из условия и просто его возвращать

Umka7389 reviewed (commented):
Молодец, хорошая работа! Замечания в комментариях

Maks1-15 committed:
7 фз
Изменения согласно правкам:
создание Enum для switch, переименование методов и переменных. Добавлено комментариев по коду
